src/oemof/solph/_models.py

```diff
@@ -110,7 +114,13 @@ def __init__(self, energysystem, **kwargs):
         # ########################  Arguments #################################
 
         self.name = kwargs.get("name", type(self).__name__)
-        self.es = energysystem
+        self.iscellular = isinstance(energysystem, dict)
+
+        if self.iscellular:
+            self.es = list(energysystem.keys())[0]
+            self.ec = energysystem[self.es]
```

Comment:
I think since you cast to a list anyways and the dict has only one key, you might make it a list rightaways and declare the 0th element to be your "master" energy system.